### ISSUE TITLE
A tool for gathering diagnostic information from the Cassandra operator and its clusters

### ISSUE THREAD

**Issue by wallrj** [sky-uk/cassandra-operator#123]
We gather various diagnostics from the cluster when E2E tests fail, but the code is buried in the E2E test suite.

It'd be nice to extract this and create a separate command line tool which users of the operator can use to gather information about the operator if they suspect it is failing.

It might also be used after a successful CI run to create a record of the build.

**Comment by sebbonnet** [comment]
That is an interesting idea: a tool for troubleshooting a cluster status.
I wonder though how useful it would be if we exposed some of this information in the CRD status subresource  - which is something we thought about some time ago, but haven't fleshed out the details (see https://github.com/sky-uk/cassandra-operator/issues/40).
I think we would want the status CRD subresource to have the number of replicas ready/not ready, pending and completed operations, perhaps even cassandra nodetool status. 
When troubleshooting I found it useful to see events for statefulset and pods and occassionally pods and operator logs. Adding these events (not the logs) to the status resource would be useful too, but I'm not sure if this type of info really fits into a `status` resource.

Publishing the travis test results would be very useful, but where to?
AWS S3 is a candidate, but it would be nice if there was a way to add the test results to the GitHub PR itself. We would need to investigate our options
